rapidfuzz/details/types.hpp

```diff
@@ -218,17 +218,21 @@ class Editops : private std::vector<EditOp> {
 public:
     using std::vector<EditOp>::size_type;
 
-    Editops() : src_len(0), dest_len(0)
+    Editops() noexcept : src_len(0), dest_len(0)
     {}
 
-    Editops(size_type count, const EditOp& value)
+    Editops(size_type count, const EditOp& value) noexcept(
+        std::is_nothrow_constructible<std::vector<EditOp>, size_type, const EditOp&>::value)
```

Comment:
Can this even be true? I do not see how the memory allocation of std::vector can be noexcept without a custom allocator.

Reply:
You are right. I double checked the STL vector implementation of MSVC as well as the specification listed in [cppreference](https://en.cppreference.com/w/cpp/container/vector/vector), this particular constructor in which memory allocation is performed does not have the noexcept specifier even if a custom allocator is noexcept.

Reply:
The comment below is restored from my deleted pending review:

Originally, I had:
```cpp
Editops(size_type count, const EditOp& value) noexcept(
        std::is_nothrow_constructible<std::vector<EditOp>, decltype(count), decltype(value)>::value)
```
It compiled fine with MSVC 2022 but failed with MSVC 2017. According to the error message, it seems MSVC 2017 compile did not expand `decltype(count)` until after within the scope of `std::is_nothrow_constructible`, thereby failing to resolve `count` and `value`. It appears to be a bug with MSVC 2017, similar (but not the same) to the bug report found here (https://developercommunity.visualstudio.com/t/non-type-template-parameter-pack-in-decltype-does/395587). Anyways, for maximal compatibility, I changed the argument types to be explicitly written.

Reply:
This is fixed.